noodles-bam/src/record/codec/encoder/sequence.rs

```diff
@@ -7,26 +7,27 @@ pub fn put_sequence<B>(dst: &mut B, read_length: usize, sequence: &Sequence) ->
 where
     B: BufMut,
 {
-    // § 1.4.10 "`SEQ`" (2022-08-22): "If not a '*', the length of the sequence must equal the sum
-    // of lengths of `M`/`I`/`S`/`=`/`X` operations in `CIGAR`."
-    if read_length > 0 && sequence.len() != read_length {
-        return Err(io::Error::new(
+    if sequence.is_empty() {
+        Ok(())
+    } else if read_length > 0 && sequence.len() != read_length {
+        // § 1.4.10 "`SEQ`" (2022-08-22): "If not a '*', the length of the sequence must equal the sum
+        // of lengths of `M`/`I`/`S`/`=`/`X` operations in `CIGAR`."
+        Err(io::Error::new(
             io::ErrorKind::InvalidInput,
-            "read length-sequence length mismatch",
-        ));
-    }
-
-    let mut bases = sequence.as_ref().iter().copied();
-
-    while let Some(l) = bases.next() {
-        // § 4.2.3 "SEQ and QUAL encoding" (2021-06-03): "When `l_seq` is odd the bottom 4 bits of
-        // the last byte are undefined, but we recommend writing these as zero."
-        let r = bases.next().unwrap_or(Base::Eq);
-        let b = encode_base(l) << 4 | encode_base(r);
-        dst.put_u8(b);
+            "read length and sequence length mismatch",
+        ))
+    } else {
+        let mut bases = sequence.as_ref().iter();
```

Comment:
Why was the [`Copied` adapter](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.copied) removed from this iterator? It removes the need to deref the values below.

Reply:
Just a silly preference of mine, changed it back.